packages/core/strapi/src/core/loaders/components.ts

```diff
@@ -60,4 +60,6 @@ export default async function loadComponents(strapi: Strapi) {
 
     return acc;
   }, {} as ComponentMap);
+
+  strapi.container.get('components').add(components);
```

Comment:
Is it using add instead of set on purpose? Since this is the component loader I woulld have expected it to clean previous state and load only those components (more could be added in later steps)
Does this mean it's ok for this load method to have side effects if called multiple times?

Honestly just asking out of curiosity here

Reply:
add is just a setMany we do this everywhere so I stay consistent